ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OKMeans.scala

```diff
@@ -0,0 +1,128 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package ai.h2o.sparkling.ml.algos
+
+import ai.h2o.sparkling.ml.params.{H2OAlgoParamsHelper, H2OAlgoUnsupervisedParams}
+import hex.kmeans.KMeansModel.KMeansParameters
+import hex.kmeans.{KMeans, KMeansModel}
+import hex.schemas.GLMV3.GLMParametersV3
+import org.apache.spark.h2o.H2OContext
+import org.apache.spark.ml.util.{DefaultParamsReadable, Identifiable}
+import org.apache.spark.sql.SparkSession
+
+/**
+  * H2O KMeans algorithm exposed via Spark ML pipelines.
+  */
+class H2OKMeans(override val uid: String) extends H2OUnsupervisedAlgorithm[KMeans, KMeansModel, KMeansParameters] with H2OKMeansParams {
+
+  def this() = this(Identifiable.randomUID("kmeans"))
+}
+
+object H2OKMeans extends DefaultParamsReadable[H2OKMeans]
+
+
+/**
+  * Parameters for Spark's API exposing underlying H2O model.
+  */
+trait H2OKMeansParams extends H2OAlgoUnsupervisedParams[KMeansParameters] {
+
+  type H2O_SCHEMA = GLMParametersV3
+
+  protected def paramTag = reflect.classTag[KMeansParameters]
+
+  protected def schemaTag = reflect.classTag[H2O_SCHEMA]
+
+  //
+  // Param definitions
+  //
+  private val maxIterations = intParam("maxIterations", "Maximum number of KMeans iterations")
+  private val standardize = booleanParam("standardize", "Standardize columns before computing distances")
+  private val init = stringParam("init", "Initialization mode")
```

Comment:
http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/init.html. Will improve the comment here, but I would like to avoid duplicating the full explanation on the web. Will point to this link in the documentation.